### PR TITLE
Various bug fixes

### PR DESCRIPTION
**Pull request by robertwoj-microsoft**
## Description

This is a short series of bugfixes that could potentially be merged to Azure/azure-osconfig, basically issues reported by address sanitizer just by running unit tests locally.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] I added unit-tests to validate my changes. All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/common/commonutils/CommandUtils.c [hunk 1]
```diff
@@ -339,7 +340,7 @@ int ExecuteCommand(void* context, const char* command, bool replaceEol, bool for
         OsConfigLogInfo(log, "Context: '%p'", context);
         OsConfigLogInfo(log, "Command: '%s'", command);
         OsConfigLogInfo(log, "Status: %d (errno: %d)", status, errno);
-        OsConfigLogInfo(log, "Text result: '%s'", (NULL != textResult) ? (*textResult) : "");
+        OsConfigLogInfo(log, "Text result: '%s'", (NULL != textResult && NULL != *textResult) ? (*textResult) : "");
```
Comment (robertwoj-microsoft) [inline src/common/commonutils/CommandUtils.c:343]: textResult may be NULL if file is empty

File: src/common/commonutils/DeviceInfoUtils.c [hunk 2]
```diff
@@ -26,7 +26,7 @@ void RemovePrefixBlanks(char* target)
         i += 1;
     }
 
-    memcpy(target, target + i, targetLength - i);
```
Comment (robertwoj-microsoft) [inline src/common/commonutils/DeviceInfoUtils.c:29]: Memory overlaps

File: src/common/commonutils/FileUtils.c [hunk 3]
```diff
@@ -309,6 +309,11 @@ bool ConcatenateFiles(const char* firstFileName, const char* secondFileName, boo
 
 int RestrictFileAccessToCurrentAccountOnly(const char* fileName)
 {
+    if (NULL == fileName)
```
Comment (robertwoj-microsoft) [inline src/common/commonutils/FileUtils.c:312]: Happens to be NULL during testing

File: src/common/commonutils/PassUtils.c [hunk 4]
```diff
@@ -81,9 +81,13 @@ int CheckEnsurePasswordReuseIsLimited(int remember, char** reason, void* log)
             g_etcPamdCommonPassword, g_etcPamdSystemAuth, g_remember);
     }
 
-    if (status && (false == FindPamModule(g_pamUnixSo, log)))
```
Comment (robertwoj-microsoft) [inline src/common/commonutils/PassUtils.c:84]: FindPamModule allocates and returns the allocated pointer

File: src/common/commonutils/SshUtils.c [hunk 5]
```diff
@@ -292,6 +292,7 @@ static int CheckOnlyApprovedMacAlgorithmsAreUsed(const char* macs, char** reason
         return status;
```
Comment (robertwoj-microsoft) [inline src/common/commonutils/SshUtils.c:292]: Leaks sshMacs

File: src/platform/tests/PlatformTests.cpp [hunk 6]
```diff
@@ -26,12 +26,7 @@ namespace Tests
         }
         else
         {
-            char* handle = new (std::nothrow) char[strlen(g_mockHandle) + 1];
```
Comment (robertwoj-microsoft) [inline src/platform/tests/PlatformTests.cpp:29]: The handle is internally released by free, not delete[]

**Review by robertwoj-microsoft — COMMENTED** [review]
Dropped some comments with explanations

File: src/common/commonutils/SshUtils.c [hunk 7]
```diff
@@ -275,7 +275,7 @@ static int IsSshConfigIncludeSupported(void* log)
 
 static int CheckOnlyApprovedMacAlgorithmsAreUsed(const char* macs, char** reason, void* log)
 {
-    char* sshMacs = DuplicateStringToLowercase(g_sshMacs);
+    char *sshMacs = NULL;
```
Comment (jepio) [inline src/common/commonutils/SshUtils.c:278]: stick to existing style ('*' location)

File: src/common/commonutils/SshUtils.c [hunk 8]
```diff
@@ -343,7 +344,7 @@ static int CheckOnlyApprovedMacAlgorithmsAreUsed(const char* macs, char** reason
 
 static int CheckAppropriateCiphersForSsh(const char* ciphers, char** reason, void* log)
 {
-    char* sshCiphers = DuplicateStringToLowercase(g_sshCiphers);
+    char *sshCiphers = NULL;
```
Comment (jepio) [inline src/common/commonutils/SshUtils.c:347]: same comment about star location

File: src/platform/MpiServer.c [hunk 9]
```diff
@@ -199,7 +199,7 @@ HTTP_STATUS HandleMpiCall(const char* uri, const char* requestBody, char** respo
     const char* client = NULL;
     const char* component = NULL;
     const char* object = NULL;
-    const char* payload = NULL;
+    char* payload = NULL;
```
Comment (kkanas) [inline src/platform/MpiServer.c:202]: Style -> char *payload = NULL;

Comment (robertwoj-microsoft) [reply src/platform/MpiServer.c:202]: This in fact would change the style

File: src/common/commonutils/SshUtils.c (hunk 7 above)
Comment (robertwoj-microsoft) [reply src/common/commonutils/SshUtils.c]: fixed

File: src/common/commonutils/SshUtils.c (hunk 8 above)
Comment (robertwoj-microsoft) [reply src/common/commonutils/SshUtils.c]: fixed

File: src/common/commonutils/PassUtils.c [hunk 10]
```diff
@@ -81,9 +81,13 @@ int CheckEnsurePasswordReuseIsLimited(int remember, char** reason, void* log)
...
+    if (status)
     {
-        OsConfigCaptureReason(reason, "The PAM module '%s' is not available. Automatic remediation is not possible", g_pamUnixSo);
+        char* pamModule = FindPamModule(g_pamUnixSo, log);
+        if(NULL == pamModule) {
```
Comment (jepio) [inline src/common/commonutils/PassUtils.c:87]: put `{` on newline

**Review by jepio — APPROVED** [review]
just that one style change

File: src/common/commonutils/PassUtils.c (hunk 10 above)
Comment (robertwoj-microsoft) [reply src/common/commonutils/PassUtils.c]: fixed

**Comment by robertwoj-microsoft** [comment]
Closed as these fixes were merged to upstream.

**Pull request closed without merging**